spec/controllers/aid_applications/finisheds_controller_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+describe AidApplications::DisbursementsController do
+  let(:supervisor) { create :supervisor }
+  let!(:aid_application) { create :aid_application, :approved }
+
+  before do
+    sign_in supervisor
+  end
+
+  context 'when Re-send Activation Code button is clicked' do
+    let!(:aid_application) { create :aid_application, :disbursed, sms_consent: false }
+
+    it 'sends resends the activation code' do
+      put :edit, params: {
+        organization_id: aid_application.organization.id,
+        aid_application_id: aid_application.id,
+        form_action: "resend_code"
+      }
+
+      expect do
+        aid_application.send_submission_notification
+      end.to have_enqueued_job(ActionMailer::MailDeliveryJob).with("ApplicationEmailer", "basic_message", "deliver_now",
```

Comment:
This expect/do block should be wrapping the running of the controller action, not sending the email directly

Reply:
Done